src/Build/Instance/ProjectInstance.cs

```diff
@@ -2035,9 +2035,15 @@ internal static ProjectInstance[] LoadSolutionForBuild(string projectFile, Prope
             else
             {
                 int solutionVersion;
-                int visualStudioVersion;
-                SolutionFile.GetSolutionFileAndVisualStudioMajorVersions(projectFile, out solutionVersion, out visualStudioVersion);
-
+                int visualStudioVersion = 0;
+                if (FileUtilities.IsSolutionFilterFilename(projectFile))
+                {
+                    solutionVersion = 16;
```

Comment:
This might age badly if there's a 17 that supports slnf and some new thing. Can you get it from the referenced `.sln`?

Reply:
Yes. My rationale is that it wouldn't make any difference now, and this way avoids file i/o (slow) and a string search. Would you be ok with it if I just put a comment like `// TODO: parse file to get accurate solution version`?